RestaurantReservation.API/Controllers/CustomersController.cs

```diff
@@ -0,0 +1,173 @@
+        private const int DefaultPageSize = 10;
+
+        /// <summary>
+        /// Constructor for Customer Controller
+        /// </summary>
+        /// <param name="customerRepository">The repository for customer data</param>
+        /// <param name="mapper">The AutoMapper instance</param>
+        /// <exception cref="ArgumentNullException">Thrown if customerRepository or mapper is null</exception>
+        public CustomersController(CustomerRepository customerRepository, IMapper mapper)
+        {
+            _customerRepository = customerRepository ?? throw new ArgumentNullException(nameof(customerRepository));
+            _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+        }
+
+        /// <summary>
+        /// Get list of customers
+        /// </summary>
+        /// <param name="pageNumber">the page number to return</param>
+        /// <param name="pageSize"> max number of records per page</param>
+        /// <returns>List of customers</returns>
+        /// <response code="200">Returns the list of customers</response>
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<CustomersWithoutReservationsDto>>> GetCustomers(int pageNumber = 1, int pageSize = DefaultPageSize)
+        {
+            var(customerEntities, paginationMetaData) = await _customerRepository.GetAllAsync(pageNumber, pageSize);
+            Response.Headers.Add("X-Pagination", JsonSerializer.Serialize(paginationMetaData));
+            return Ok(_mapper.Map<IEnumerable<CustomersWithoutReservationsDto>>(customerEntities));
+        }
+
+        /// <summary>
+        /// Get customer by Id
+        /// </summary>
+        /// <param name="id">id of customer to return</param>
+        /// <returns>Customer</returns>
+        /// <response code="200">Returns the customer with the specified id</response>
+        /// <response code="404">If no customer with the specified id is found</response>
+        [HttpGet("{id}", Name = "GetCustomer")]
+        public async Task<IActionResult> Get(int id)
+        {
+            var customer = await _customerRepository.GetByIdAsync(id); 
+            if(customer == null)
+            {
+                var error = CustomerErrors.NotFound(id);
+                return NotFound(error);  
+            }
+
+            return Ok(_mapper.Map<CustomersWithoutReservationsDto>(customer));
+        }
+
+        /// <summary>
+        /// create new customer
+        /// </summary>
+        /// <param name="customerDto">Customer data</param>
+        /// <returns>The created customer</returns>
+        /// <response code="201">Returns the created customer</response>
+        /// <response code="409">If  customer email isn't unique</response>
+        [HttpPost]
+        public async Task<ActionResult<CustomerDto>> Create(CustomerDto customerDto)
+        {
+            var existingCustomer = await _customerRepository.GetByEmailAsync(customerDto.Email);
```

Comment:
maybe it's better to have a function that check if this email is existed or not rather getting the user class, or at least encapsulate this inside such funtion. - will explain.